apps/web/src/server/lib/sas-token.js

```diff
@@ -0,0 +1,31 @@
+import {
+	AccountSASPermissions,
+	AccountSASResourceTypes,
+	AccountSASServices,
+	generateAccountSASQueryParameters,
+	SASProtocol,
+	StorageSharedKeyCredential
+} from '@azure/storage-blob';
+import config from '../../../environment/config.js';
+
+const createSasToken = async () => {
+	const sharedKeyCredential = new StorageSharedKeyCredential(
```

Comment:
We can actually use the same blob credential that we use for the end user to upload files here, and create the SAS token using the user account credentials here: https://learn.microsoft.com/en-us/azure/storage/common/storage-sas-overview#user-delegation-sas

Reply:
done and refactored the upload function as well. now there's an external functoin that return an AD token starting from the web server session